torch/csrc/autograd/profiler.h

```diff
@@ -138,9 +141,35 @@ struct TORCH_API Event final {
         handle_(handle),
         shapes_(shapes) {
     record(record_cuda);
+    p_cuda_event_counter = std::make_shared<unsigned>(1);
```

Comment:
ideally we'd want to avoid any unnecessary memory allocations, e.g. if cuda is not used